app/assets/javascripts/live_search.js

```diff
@@ -382,10 +378,19 @@
       xhr.addEventListener('load', done)
       xhr.send()
     } else {
+      this.xmlHttpDone()
       this.displayResults(cachedResultData, searchState)
     }
   }
 
+  LiveSearch.prototype.xmlHttpDone = function () {
```

Comment:
I think this name is a little misleading as it can be called when no XHR has been made.

I can't think of a particularly better name for the method, since it seems to do a few things, but I did wonder if it's actually all part of displaying results (since they seem to be coupled) and should be a part of that method.

Reply:
Thanks for this comment - it's always worth remembering that code should be readable for the next developer. With that in mind, I've split this into two functions, one for the URL change and one for calling the tracking. It feels a little verbose to do this, but I think it makes the code clearer in terms of what it's actually doing.